### ISSUE TITLE
multi worker training in tensorflow gives error - "terminate called without an active exception aborted"

### ISSUE THREAD

**Issue by ranvirdesai** [tensorflow/tensorflow#55250]
I am running a tensorflow distributed training model on multiple workers using vertex AI/local and multiworkermirroredstrategy

After completion of all epochs it gives the error of "terminate called without an active exception aborted"

python=3.8
tensorflow=2.7

Can anyone explain why is this happening.

**Comment by tilakrayal** [comment]
@ranvirdesai ,
Can you please take a look at this [SO link](https://stackoverflow.com/questions/7381757/c-terminate-called-without-an-active-exception) and [issue](https://github.com/tensorflow/tensorflow/issues/51768) with the similar error.It helps.Thanks!

**Comment by ranvirdesai** [comment]
@tilakrayal 
I face issue at the end of training, after completion of epochs and I am not using tensorboard profiling.

Mirroredstrategy and TPUStrategy works fine.

Just the Multiworkermirroredstrategy fails giving error - "terminate called without an active exception aborted"

**Comment by tilakrayal** [comment]
@ranvirdesai ,
In order to expedite the trouble-shooting process, could you please provide the complete code to reproduce the issue reported here.

**Comment by ranvirdesai** [comment]
@tilakrayal 

import tensorflow as tf
import tensorflow_datasets as tfds

print("tf version", tf.__version__)

strategy = tf.distribute.MultiWorkerMirroredStrategy()

(ds_train, ds_test), ds_info = tfds.load(
    'mnist',
    split=['train', 'test'],
    shuffle_files=True,
    as_supervised=True,
    with_info=True,
)

def normalize_img(image, label):
  """Normalizes images: `uint8` -> `float32`."""
  return tf.cast(image, tf.float32) / 255., label

ds_train = ds_train.map(
    normalize_img, num_parallel_calls=tf.data.AUTOTUNE)
ds_train = ds_train.cache()
ds_train = ds_train.shuffle(ds_info.splits['train'].num_examples)
ds_train = ds_train.batch(128)
ds_train = ds_train.prefetch(tf.data.AUTOTUNE)


ds_test = ds_test.map(
    normalize_img, num_parallel_calls=tf.data.AUTOTUNE)
ds_test = ds_test.batch(128)
ds_test = ds_test.cache()
ds_test = ds_test.prefetch(tf.data.AUTOTUNE)

def build_cnn_model():
    return tf.keras.Sequential([
      tf.keras.Input(shape=(28, 28)),
      tf.keras.layers.Reshape(target_shape=(28, 28, 1)),
      tf.keras.layers.Conv2D(32, 3, activation='relu'),
      tf.keras.layers.Flatten(),
      tf.keras.layers.Dense(128, activation='relu'),
      tf.keras.layers.Dense(10)
    ])

with strategy.scope():
    model = build_cnn_model()
    model.compile(
        optimizer=tf.keras.optimizers.Adam(0.001),
        loss=tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True),
        metrics=[tf.keras.metrics.SparseCategoricalAccuracy()],
    )

model.fit(
    ds_train,
    epochs=2,
    validation_data=ds_test,
)


It is simple mnist model which gives error.
Also model training on multtiple CPU machines works fine but when we attach GPUs to it gives the error - 

**terminate called without an active exception aborted**

Thanks

**Comment by tilakrayal** [comment]
@ranvirdesai ,
Can you please confirm whether this issue is duplicate of #55178?.Thanks!

**Comment by fig666** [comment]
The model isn't duplicated or rescripted in any form but could be a test modification plugin. Are you having any complications with running it?

**Comment by fig666** [comment]
Are you trying to mass script or group the code plugin for a infinite loop or machine learning module?

**Comment by ranvirdesai** [comment]
I was using a small dataset and it was giving problems, but when I used bigger dataset it worked.
Also changed python 3.8 to 3.7

**Comment by fig666** [comment]
for the time being i am satisfied with the resolution outlook of python.
